### PR TITLE
support set init tokens by user for sanas 

### PR DESCRIPTION
**Pull request by ceci3**
1. random generate init token         ✔︎
2. support set init token by user     ✔︎
3. set  temperature correlate with client num    ✔︎
4. set temperature correlate with init token set by user or not ✔︎
5. fix the bug in self._current_token   ✔︎
6. remove client_name in server if a client not request next tokens in double train one tokens' time. ✔︎

### CODE REVIEW

File: paddleslim/nas/sa_nas.py [hunk 1]
```diff
@@ -64,7 +67,11 @@ def __init__(self,
         self._init_temperature = init_temperature
         self._is_server = is_server
         self._configs = configs
-        self._key = hashlib.md5(str(self._configs).encode("utf-8")).hexdigest()
+        self._init_tokens = init_tokens
+        self._client_name = hashlib.md5(str(time.time()).encode(
```
Comment (wanghaoshuang) [inline paddleslim/nas/sa_nas.py]: 如果两个client同时启动，名字是不是就一样了？加入一个随机信息？

File: paddleslim/common/controller_client.py [hunk 2]
```diff
@@ -26,7 +26,11 @@ class ControllerClient(object):
     Controller client.
     """
 
-    def __init__(self, server_ip=None, server_port=None, key=None):
+    def __init__(self,
+                 server_ip=None,
+                 server_port=None,
+                 key=None,
+                 client_name=None):
```
Comment (wanghaoshuang) [inline paddleslim/common/controller_client.py:33]: 注意更新下注释

File: paddleslim/common/sa_controller.py [hunk 3]
```diff
@@ -34,7 +34,7 @@ class SAController(EvolutionaryController):
     def __init__(self,
                  range_table=None,
                  reduce_rate=0.85,
-                 init_temperature=1024,
+                 init_temperature=None,
```
Comment (wanghaoshuang) [inline paddleslim/common/sa_controller.py:37]: 注意更新注释

File: paddleslim/nas/sa_nas.py (hunk 1 above)
Comment (ceci3) [reply paddleslim/nas/sa_nas.py]: done

File: paddleslim/common/controller_client.py (hunk 2 above)
Comment (ceci3) [reply paddleslim/common/controller_client.py:33]: done

File: paddleslim/common/sa_controller.py (hunk 3 above)
Comment (ceci3) [reply paddleslim/common/sa_controller.py:37]: done

**Review by wanghaoshuang — APPROVED** [review]

**Pull request merged**